MVP for Power Curves and Energy Production

sameersk2k opened the pull request:
this connects #41
Changes:
1. athena_data _fetcher : added parameter **get_d**f to fetch full dataframe if needed as in case for energy production calculation.
2. wtk_data_controller: added endpoints **available_powercurves** for dynamically fetching available power curves and filling the settings component in UI and **energy_production** to calculate it and return to UI right pane component.
3. Added **power_curve** module under app.
power_curve module has **power_curve_manager** that initializes objects for different power_curves and has methods for energy production calculation. it also has powercurve.py that handles conversion logics.

RLiNREL commented on windwatts-api/app/controllers/wtk_data_controller.py:
```diff
@@ -28,11 +30,14 @@
 
 
-@router.get("/windspeed/{avg_type}", summary="Retrieve wind speed with avg type - wtk data")
+# @router.get("/windspeed/{avg_type}", summary="Retrieve wind speed with avg type - wtk data")
```
please don't comment this line, it is needed for `avg_type`

RLiNREL commented on windwatts-api/app/controllers/wtk_data_controller.py:
```diff
@@ -61,4 +66,35 @@ def get_windspeed(lat: float, lng: float, height: int, avg_type: str = 'global',
+
+@router.get("/available_powercurves", summary="Fetch all available power curves")
+def fetch_available_powercurves():
+    "returns available power curves"
```
is this line a comment?

RLiNREL commented on windwatts-api/app/controllers/wtk_data_controller.py:
```diff
@@ -61,4 +66,35 @@ def get_windspeed(lat: float, lng: float, height: int, avg_type: str = 'global',
+            "lat": lat,
+            "lng": lng,
+            "height": None,
+            "avg_type" : None,
```
see the comments in the athena fetcher, let's minimize the inputs `get_df`, but use the avg_type == 'none' to determine if the non averaged df to be returned.

RLiNREL commented on windwatts-api/app/controllers/wtk_data_controller.py:
```diff
@@ -61,4 +66,35 @@ def get_windspeed(lat: float, lng: float, height: int, avg_type: str = 'global',
+    energy_production_df = power_curve_manager.fetch_energy_production_df(df, height, selected_powercurve)
+    print(f"nrel-reference-{selected_powercurve}-kW")
+
+    yearly_avg_energy_production = power_curve_manager.fetch_yearly_avg_energy_production(energy_production_df,height)
```
could design to return the result directly instead of call two methods `fetch_energy_production_df` and `fetch_yearly_avg_energy_production`

RLiNREL reviewed (commented):
Overall the added power curve/energy production is working, but need some clean ups later.
A couple of quick changes are:
- uncomment the `# @router.get("/windspeed/{avg_type}", summary="Retrieve wind speed with avg type - wtk data")`
- remove the `get_df` input for fetch_data, but check `avg_type` if the original df without averaging should be returned.

will merge the changes first when these two changes are made.

sameersk2k commented:
- removed get_df param from the athena_data_fetcher.fetch_data.
- included avg_type="none" condition to fetch full data frame.
- uncommented the router /windspeed/{avg_type}

sameersk2k commented:
- modified energy_production function to make single call to fetch computed production instead of 2( 1 to fetch df and 1 to compute yearly averages).
- added exception handling in energy_production
- made var names smaller in power_curve_manager

sameersk2k commented:
- Updates return types for power_curve_manager.fetch_yearly_avg_energy_production and power_curve_manager.fetch_monthly_avg_energy_production from data frame to dict.
- Added documentation for the dict format within the above functions.

RLiNREL reviewed (approved):
MVP for power curve/production computation for #41 with more re-organization and clean up to come later in separate issues.
- The functionalities are encapsulated in the classes and methods under /powe_curves folder.
- The endpoint is "/energy_production" with return type key-value pair of {"energy_production": value}
